src/contracts/Test/IbexaTestKernel.php

```diff
@@ -0,0 +1,207 @@
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Core\Test;
+
+use Doctrine\Bundle\DoctrineBundle\DoctrineBundle;
+use Doctrine\DBAL\Connection;
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ChainConfigResolver;
+use eZ\Bundle\EzPublishCoreBundle\EzPublishCoreBundle;
+use eZ\Bundle\EzPublishLegacySearchEngineBundle\EzPublishLegacySearchEngineBundle;
+use eZ\Publish\API\Repository;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\SPI\Persistence\TransactionHandler;
+use FOS\JsRoutingBundle\FOSJsRoutingBundle;
+use JMS\TranslationBundle\JMSTranslationBundle;
+use Liip\ImagineBundle\LiipImagineBundle;
+use Psr\Log\Test\TestLogger;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\SecurityBundle\SecurityBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * Baseline test kernel that dependent packages can extend for their integration tests.
+ *
+ * ## Configuring the kernel
+ *
+ * It automatically exposes all Repository-based services for consumption in tests (marking them as public prevents
+ * them from being removed from test container). A minimal configuration Symfony framework configuration is provided,
+ * along with Doctrine connection.
+ *
+ * To supply a different configuration, extend IbexaTestKernel::loadConfiguration() method.
+ *
+ * You can supply your own services (which is something you probably want) by extending IbexaTestKernel::loadServices().
+ *
+ * If you need even more control over how the container is built you can do that by extending the
+ * IbexaTestKernel::registerContainerConfiguration().
+ *
+ * ## Adding bundles
+ *
+ * Bundles can be added by extending IbexaTestKernel::registerBundles() method (just like in any Kernel).
+ *
+ * ## Exposing your services
+ *
+ * To add services to the test Kernel and make them available in tests via IbexaKernelTestCase::getServiceByClassName(),
+ * you'll need to extend IbexaTestKernel::getExposedServicesByClass() and / or IbexaTestKernel::getExposedServicesById()
+ * method.
+ *
+ * IbexaTestKernel::getExposedServicesByClass() is a simpler variant provided for services that are registered in
+ * service container using their FQCN.
+ *
+ * IbexaTestKernel::getExposedServicesById() is useful if your service is not registered as it's FQCN (for example,
+ * if you have multiple services for the same class / interface).
+ *
+ * If don't need the repository services (or not all), you can replace the IbexaTestKernel::EXPOSED_SERVICES_BY_CLASS and
+ * IbexaTestKernel::EXPOSED_SERVICES_BY_ID consts in extending class, without changing the methods above.
+ */
```

Comment:
Please move it to documentation if it's meant to be used in public or into our internal wiki.

Reply:
I've discussed this a bit with @alongosz and I think we should mark this class as `@experimental` (since it's possible something will change) and the phpdoc description should be copied into internal wiki.

I'd rather keep the class usage description in the class and consider it the source of truth for any documentation that follows, be it internal or official. WDYT?

Reply:
@adamwojs documentation added to internal wiki.